packages/protocol/contracts/governance/Election.sol

```diff
@@ -327,6 +327,52 @@ contract Election is
     return true;
   }
 
+  /**
+   * @notice Revokes `value` pending or active votes for `group`. First revokes all pending votes
+   *         and, if `value` haven't been revoked yet, revokes additional active votes.
+   *         Basically calls `revokePending` and `revokeActive` but only resorts groups once.
+   * @param group The validator group to revoke votes from.
+   * @param value The number of votes to revoke.
+   * @param lesser The group receiving fewer votes than the group for which the vote was revoked,
+   *               or 0 if that group has the fewest votes of any validator group.
+   * @param greater The group receiving more votes than the group for which the vote was revoked,
+   *                or 0 if that group has the most votes of any validator group.
+   * @param index The index of the group in the account's voting list.
+   * @return uint256 Number of votes successfully revoked, with a max of `value`.
+   */
+  function _revokeVotes(
```

Comment:
Emit an event?

Reply:
An event `ValidatorGroupVoteRevoked` is emitted if any votes are successfully decremented. This is the same event as in `revokePending` and `revokeActive`. I don't think we need an additional event, let me know if you think otherwise or you just didn't see the existing event